tests/unit/fancyrag/kg/test_pipeline.py

```
@@ -222,6 +260,89 @@ def test_run_pipeline_validates_chunk_size(monkeypatch, tmp_path):
+
+def test_semantic_retry_on_format_error(monkeypatch, tmp_path):
+    calls: list[str] = []
+
+    class DummyExtractor:
+        def __init__(self, *_, **__):
+            pass
+
+        async def extract_for_chunk(self, _schema, _examples, chunk):
+            calls.append(chunk.uid)
+            if len(calls) == 1:
+                raise pipeline.LLMGenerationError("bad format")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)